SilMock/Google/Service/Directory/Resource/Groups.php

```diff
@@ -20,23 +21,52 @@ public function delete(string $groupKey)
         $groupRecords = $this->getRecords();
         foreach ($groupRecords as $groupRecord) {
             $groupRecordData = json_decode($groupRecord['data'], true);
-            if ($groupRecordData['email'] === $groupKey) {
+            $keysToCheck = $groupRecordData['aliases'];
+            $keysToCheck[] = $groupRecordData['email'];
+            if (in_array($groupKey, $keysToCheck)) {
                 $this->deleteRecordById($groupRecord['id']);
+                $mockGroupsAliasesObject = new GroupsAliases($this->dbFile);
+                foreach ($mockGroupsAliasesObject->getRecords() as $aliasRecord) {
+                    $aliasRecordData = json_decode($aliasRecord['data'], true);
+                    if ($aliasRecordData['primaryEmail'] === $groupRecordData['email']) {
+                        $mockGroupsAliasesObject->deleteRecordById($aliasRecord['id']);
+                    }
+                }
```

Comment:
It might be worth extracting this as a `deleteAliasesForGroup()` protected function of some sort. Just an idea.